vss-accessor/tests/tests.rs

```diff
@@ -0,0 +1,272 @@
+			message: "InvalidRequestException".to_string(),
+		};
+		let mock_server =
+			mockito::mock("POST", Matcher::Any).with_status(400).with_body(&error_response.encode_to_vec()).create();
+
+		let get_result = vss_accessor.get("store", "key1").await;
+		assert_eq!(get_result.unwrap_err(), VssError::InvalidRequestError(ErrorResponse::default()));
+
+		let put_result = vss_accessor.put("store", Some(4), "k1", 2, b"k1v3").await;
+		assert_eq!(put_result.unwrap_err(), VssError::InvalidRequestError(ErrorResponse::default()));
+
+		let list_result = vss_accessor.list_key_versions("store", "k", Some(5), None).await;
+		assert_eq!(list_result.unwrap_err(), VssError::InvalidRequestError(ErrorResponse::default()));
+
+		// Verify 3 requests hit the server
+		mock_server.expect(3).assert();
+	}
+
+	#[tokio::test]
+	async fn test_conflict_err_handling() {
+		let base_url = mockito::server_url();
+		let vss_accessor = VssAccessor::new(&base_url).unwrap();
+
+		// Conflict Error
+		let error_response =
+			ErrorResponse { error_code: ErrorCode::ConflictException.into(), message: "ConflictException".to_string() };
+		let mock_server =
+			mockito::mock("POST", Matcher::Any).with_status(409).with_body(&error_response.encode_to_vec()).create();
+
+		let put_result = vss_accessor.put("store", Some(4), "k1", 2, b"k1v3").await;
+		assert_eq!(put_result.unwrap_err(), VssError::ConflictError(ErrorResponse::default()));
+
+		// Verify 1 requests hit the server
+		mock_server.expect(1).assert();
+	}
+
+	#[tokio::test]
+	async fn test_internal_server_err_handling() {
+		let base_url = mockito::server_url();
+		let vss_accessor = VssAccessor::new(&base_url).unwrap();
+
+		// Internal Server Error
+		let error_response = ErrorResponse {
+			error_code: ErrorCode::InternalServerException.into(),
+			message: "InternalServerException".to_string(),
+		};
+		let mock_server =
+			mockito::mock("POST", Matcher::Any).with_status(500).with_body(&error_response.encode_to_vec()).create();
+
+		let get_result = vss_accessor.get("store", "key1").await;
+		assert_eq!(get_result.unwrap_err(), VssError::InternalServerError(ErrorResponse::default()));
+
+		let put_result = vss_accessor.put("store", Some(4), "k1", 2, b"k1v3").await;
+		assert_eq!(put_result.unwrap_err(), VssError::InternalServerError(ErrorResponse::default()));
+
+		let list_result = vss_accessor.list_key_versions("store", "k", Some(5), None).await;
+		assert_eq!(list_result.unwrap_err(), VssError::InternalServerError(ErrorResponse::default()));
+
+		// Verify 3 requests hit the server
+		mock_server.expect(3).assert();
+	}
+
+	#[tokio::test]
+	async fn test_internal_err_handling() {
+		let base_url = mockito::server_url();
+		let vss_accessor = VssAccessor::new(&base_url).unwrap();
+
+		let error_response = ErrorResponse { error_code: 999, message: "UnknownException".to_string() };
+		let mut mock_server =
+			mockito::mock("POST", Matcher::Any).with_status(999).with_body(&error_response.encode_to_vec()).create();
```

Comment:
i don't see this warning. 
is it a compiler nuance ? because the value is being mutated and also dropped explicitly using that variable. 
Or do you suggest we add an "_" in spite of that?